src/Playwright/Core/Frame.cs

```diff
@@ -195,6 +195,11 @@ public async Task WaitForLoadStateAsync(LoadState? state = default, FrameWaitFor
 
         public async Task<IResponse> WaitForNavigationAsync(FrameWaitForNavigationOptions options = default)
         {
+            if (Page.IsClosed)
```

Comment:
If you go this path, you need to add 300 checks like this in all API methods. The point is that underlying transport error is bubbled every time the call is made after connection is closed.